### ISSUE TITLE
"This authentication method is not suported by the BitBucket Cloud Rest API" log in BB Server client

### ISSUE THREAD

**Issue by uowiec** [jenkinsci/bitbucket-push-and-pull-request-plugin#148]
While having only repo on BitBucket Server on prem, the following log is visible when a PR is opened and build triggered:

```
2020-11-23 09:36:38.019+0000 [id=18101] INFO    i.j.p.b.BitBucketPPRTrigger#scheduleJob: SCM changes detected in check_pull_request. Triggering  #71
2020-11-23 09:36:45.858+0000 [id=18101] INFO    i.j.p.b.o.BitBucketPPRObservable#notifyObservers: Event: BitBucketPPRBuildStarted [context=BitBucketPPREventContext [action=BitBucketPPRPullRequestServerAction, credentialsId=****, filter=io.jenkins.plugins.bitbucketpushandpullrequest.filter.pullrequest.server.BitBucketPPRPullRequestServerTriggerFilter@31734282, run=null, scmTrigger=hudson.plugins.git.GitSCM@3e43c591, url=ssh://******, userRemoteConfig=null => ssh://***** (null)], handler=null] for observer io.jenkins.plugins.bitbucketpushandpullrequest.observer.BitBucketPPRPullRequestServerObserver@611b7b36
2020-11-23 09:36:45.858+0000 [id=18101] WARNING i.j.p.b.o.BitBucketPPRPullRequestServerObserver#callClient2: This authentication method is not suported by the BitBucket Cloud Rest API.
2020-11-23 09:36:45.861+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:45.875+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:46.787+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:46.824+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:46.826+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:46.847+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:46.848+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:53.271+0000 [id=18096] WARNING i.j.p.b.e.BitBucketPPREnvironmentContributor#lambda$buildEnvironmentFor$0: Something didnt work: {0}
2020-11-23 09:36:53.280+0000 [id=18101] INFO    i.j.p.b.o.BitBucketPPRObservable#notifyObservers: Event: BitBucketPPRBuildFinished [context=BitBucketPPREventContext [action=BitBucketPPRPullRequestServerAction, credentialsId=***, filter=io.jenkins.plugins.bitbucketpushandpullrequest.filter.pullrequest.server.BitBucketPPRPullRequestServerTriggerFilter@31734282, run=check_pull_request #71, scmTrigger=hudson.plugins.git.GitSCM@3e43c591, url=ssh://*******, userRemoteConfig=null => ssh://***** (null)], handler=null] for observer io.jenkins.plugins.bitbucketpushandpullrequest.observer.BitBucketPPRPullRequestServerObserver@611b7b36
2020-11-23 09:36:53.280+0000 [id=18101] WARNING i.j.p.b.o.BitBucketPPRPullRequestServerObserver#callClient: This authentication method is not suported by the BitBucket Cloud Rest API.
```
The log strings can be found in BitBucketPPRClientCloudVisitor.java and BitBucketPPRClientServerVisitor.java - probably as a result of copy+paste.
No idea about which authentication method it is complaining about. The authentication to BB Server, which this job uses to pull from git is user+ssh private key. Besides the messages being logged, there are no authentication related issues, and the job (mostly) works fine.

**Comment by cdelmonte-zg** [comment]
@uowiec Which version of the plugin are you using?

**Comment by cdelmonte-zg** [comment]
Are you using ssh or https for git?
The propagation works only is oyu use the https credentials

**Comment by uowiec** [comment]
Jenkins 2.249.3
BPPRR 2.6.0
I use ssh with private key authentication. I don't use any feature which would require the authentication to BB, besides the usual SCM refresh.

**Comment by cdelmonte-zg** [comment]
@uowiec yes, the log say that ssh credentials cannot be used to communicate with the BB Cloud / Server REST API, as written in the plugin documentation.

Sorry, maybe I'm missing something, but which is exactly your issue?

**Comment by uowiec** [comment]
@cdelmonte-zg There are 2 issues I spotted so far, this one and the other one with "something didnt work" message, for which I've opened another issue. Generally the plugin is giving me hard times, I try to eliminate the spotted issues one by one hoping that things will clear out. I can't explain what exactly is going on, since there is no reliable logging or logs are missleading, and that's the only help users can get. In this case, the plugin logs "This authentication method is not suported by the BitBucket Cloud..." with the "warning" severity. It looks pretty serious at first glance, but on the other hand it mentions BB Cloud, which I don't use, and even if it was mentioning BB Server (which I use) - it still leaves me confused, since I don't need to authenticate to BB from this plugin for anyting. Also, I think the real issue here is the servicability in general.
